homeassistant/components/unifiprotect/entity.py

```diff
@@ -203,37 +196,32 @@ async def async_update(self) -> None:
 
     @callback
     def _async_set_device_info(self) -> None:
-        self._attr_device_info = DeviceInfo(
-            name=self.device.display_name,
-            manufacturer=DEFAULT_BRAND,
-            model=self.device.type,
-            via_device=(DOMAIN, self.data.api.bootstrap.nvr.mac),
-            sw_version=self.device.firmware_version,
-            connections={(dr.CONNECTION_NETWORK_MAC, self.device.mac)},
-            configuration_url=self.device.protect_url,
-        )
+        """Set device info."""
 
     @callback
-    def _async_update_device_from_protect(self, device: ProtectModelWithId) -> None:
+    def _async_update_device_from_protect(self, device: ProtectDeviceType) -> None:
         """Update Entity object from Protect device."""
-        if TYPE_CHECKING:
-            assert isinstance(device, ProtectAdoptableDeviceModel)
-
-        if last_update_success := self.data.last_update_success:
+        was_available = self._attr_available
+        if last_updated_success := self.data.last_update_success:
             self.device = device
 
-        async_get_ufp_enabled = self._async_get_ufp_enabled
-        self._attr_available = (
-            last_update_success
-            and (
-                device.state is StateType.CONNECTED
-                or (not device.is_adopted_by_us and device.can_adopt)
+        if device.model is ModelType.NVR:
+            available = last_updated_success
+        else:
```

Comment:
Would be nice to clean this up in some more in a future PR. I deferred doing that here because it would make this PR balloon in size